ui/lib/utils.ts

```diff
@@ -27,11 +27,17 @@ export function poller(cb, interval) {
   return setInterval(cb, interval);
 }
 
-export function isHTTP(uri: string) {
-  return uri.includes("http");
+export function isHTTP(uri: string): boolean {
+  // Regex from Diego Perini's gist: https://gist.github.com/dperini/729294
+  // It works better than other regular expressions for validating HTTP and HTTPS URLs.
+  const regex = new RegExp(
+    /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z0-9\u00a1-\uffff][a-z0-9\u00a1-\uffff_-]{0,62})?[a-z0-9\u00a1-\uffff]\.)+(?:[a-z\u00a1-\uffff]{2,}\.?))(?::\d{2,5})?(?:[/?#]\S*)?$/i
```

Comment:
The regular expression being used here allows the "ftp" prefix too.
It can be improved by replace `(?:https?|ftp)` with `(?:https?)`.